File: leetcode/leetcode/src/com/lintcode/countAndSay.java
package com.lintcode;

/**
 * Created by shiyanghuang on 15/5/27.
 */
public class countAndSay {
    public String countAndSay(int n) {
        // Write your code here
        String s = "1";
        while (n > 1) {
            int i = 0;
            String subs = "";
            while (i < s.length()) {
                int length = s.length();
                char c = s.charAt(i++);
                int count = 1;
                while (i < s.length() && s.charAt(i) == c) {
                    count ++;
                    i++;
                }
                subs = subs + count + c;
            }
            s = subs;
            n--;
        }
        return s;
    }

    public static void main(String[] args) {
        System.out.print(new countAndSay().countAndSay(5));
    }
}

File: leetcode/Count and Say.cpp
#include<iostream>
#include<string>
using namespace std;

string countAndSay(int n) {
    n--;
    string s1,s2;
    s1="1";
    s2="";
    while(n--)
    {
        int length=s1.length();
        int i=0;
        while(i<length)
        {
            char c=s1[i];
            int count=0;
            while(s1[i]==c&&i<length)
            {
                count++;
                i++;
            }
            s2+=char(count+'0');
            s2+=c;
        }
        s1=s2;
        s2="";
    }    
    return s1;           
}
main()
{
    for(int i=1;i<10;i++)
    cout<<countAndSay(i)<<endl;
    system("pause");
}    
